tests/Unit/ControlSystem/Test_Measurements.cpp

```diff
@@ -18,70 +21,122 @@
                                  domain::ObjectLabel::A>::Excision,
                              control_system::protocols::Submeasurement>);
-
 static_assert(
     tt::assert_conforms_to_v<control_system::measurements::CharSpeed<
                                  domain::ObjectLabel::A>::Horizon,
                              control_system::protocols::Submeasurement>);
-
+static_assert(tt::assert_conforms_to_v<
+              control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+                  Excision::interpolation_target_tag<example_list>,
+              intrp::protocols::InterpolationTargetTag>);
+static_assert(tt::assert_conforms_to_v<
+              control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+                  Horizon::interpolation_target_tag<example_list>,
+              intrp::protocols::InterpolationTargetTag>);
 static_assert(
-    tt::assert_conforms_to_v<
-        control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
-            Excision::interpolation_target_tag<
-                tmpl::list<control_system::TestHelpers::ExampleControlSystem>>,
-        intrp::protocols::InterpolationTargetTag>);
-
+    not control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+        Excision::event<example_list>::factory_creatable);
 static_assert(
-    tt::assert_conforms_to_v<
-        control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
-            Horizon::interpolation_target_tag<
-                tmpl::list<control_system::TestHelpers::ExampleControlSystem>>,
-        intrp::protocols::InterpolationTargetTag>);
+    not control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+        Horizon::event<example_list>::factory_creatable);
 
 namespace {
+void test_serialization() {
+  using EventPtr = std::unique_ptr<::Event>;
+  using non_factory_creatable_event =
+      control_system::measurements::BothHorizons::FindHorizon<
+          ::domain::ObjectLabel::A>::event<example_list>;
+  using factory_creatable_event =
+      typename non_factory_creatable_event::factory_creatable_class;
+
+  register_classes_with_charm<non_factory_creatable_event,
+                              factory_creatable_event>();
+
+  const EventPtr non_factory_event =
+      std::make_unique<non_factory_creatable_event>();
+  const EventPtr factory_event = std::make_unique<factory_creatable_event>();
+
+  const EventPtr serialized_non_factory_event =
+      serialize_and_deserialize(non_factory_event);
+  const EventPtr serialized_factory_event =
+      serialize_and_deserialize(factory_event);
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpotentially-evaluated-expression"
+#endif  // __clang__
+  CHECK(typeid(*serialized_non_factory_event.get()) ==
+        typeid(*non_factory_event.get()));
+  CHECK(typeid(*serialized_factory_event.get()) ==
+        typeid(*factory_event.get()));
```

Comment:
This is OK, but why all the redundant `.get()` calls?

Reply:
🤷🏻

Artifact of previous changes